### ISSUE TITLE
vm templating consume much more memory than without vm templating

### ISSUE THREAD

**Issue by caiwenzh** [kata-containers/kata-containers#4766]
# Description of problem

vm templating consume much more memory than without vm templating. 

Test steps:

`1.` prepare yaml file vmt.yaml

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: bbox
spec:
  selector:
    matchLabels:
      app: bbox
  replicas: 100
  template:
    metadata:
      labels:
        app: bbox
    spec:
      runtimeClassName: kata
      containers:
      - name: busybox
        image: busybox
        command: ["top"]
```

`2.` create 100 pod without vm templating.

```
[root@cloud-dev-enzo work]# echo 3  > /proc/sys/vm/drop_caches
[root@cloud-dev-enzo work]# free -h
              total        used        free      shared  buff/cache   available
Mem:          187Gi        92Gi        88Gi       4.9Gi       6.0Gi        88Gi
Swap:            0B          0B          0B
[root@cloud-dev-enzo work]# kubectl apply -f vmt.yaml
[root@cloud-dev-enzo work]# free -h
              total        used        free      shared  buff/cache   available
Mem:          187Gi       101Gi        68Gi        15Gi        17Gi        68Gi
Swap:            0B          0B          0B
```

`3.`  create 100 pod with vm templating.

```
[root@cloud-dev-enzo work]# echo 3  > /proc/sys/vm/drop_caches
[root@cloud-dev-enzo work]# free -h
              total        used        free      shared  buff/cache   available
Mem:          187Gi        92Gi        88Gi       4.9Gi       6.1Gi        88Gi
Swap:            0B          0B          0B
[root@cloud-dev-enzo work]# kubectl apply -f vmt.yaml
[root@cloud-dev-enzo work]# free -h
              total        used        free      shared  buff/cache   available
Mem:          187Gi       101Gi        28Gi        55Gi        57Gi        28Gi
Swap:            0B          0B          0B
```

# Expected result

vm templating consume less memory than without vm templating

# Actual result

vm templating consume much more memory than without vm templating

**Comment by caiwenzh** [comment]
It is caused by some misconfiguration.  The result after correct the configuration:
[root@cloud-dev-enzo work]# echo 3 > /proc/sys/vm/drop_caches
[root@cloud-dev-enzo work]# free -h
              total        used        free      shared  buff/cache   available
Mem:          187Gi        92Gi        88Gi       5.4Gi       6.5Gi        87Gi
Swap:            0B          0B          0B
[root@cloud-dev-enzo work]# kubectl apply -f vmt.yaml
deployment.apps/bbox created
[root@cloud-dev-enzo work]# free -h
              total        used        free      shared  buff/cache   available
Mem:          187Gi       101Gi        78Gi       5.4Gi       6.8Gi        78Gi
Swap:            0B          0B          0B

**Comment by XueChe** [comment]
> It is caused by some misconfiguration. The result after correct the configuration: [root@cloud-dev-enzo work]# echo 3 > /proc/sys/vm/drop_caches [root@cloud-dev-enzo work]# free -h total used free shared buff/cache available Mem: 187Gi 92Gi 88Gi 5.4Gi 6.5Gi 87Gi Swap: 0B 0B 0B [root@cloud-dev-enzo work]# kubectl apply -f vmt.yaml deployment.apps/bbox created [root@cloud-dev-enzo work]# free -h total used free shared buff/cache available Mem: 187Gi 101Gi 78Gi 5.4Gi 6.8Gi 78Gi Swap: 0B 0B 0B

I had the same problem with mem-prealloc  mode. what is the misconfiguration，thanks?
